src/Nethermind/Nethermind.Shutter/ShutterApi.cs

```diff
@@ -92,11 +103,11 @@ TimeSpan slotLength
 
         KeyValidator = new ShutterKeyValidator(_cfg, Eon, logManager);
 
-        InitP2P(_cfg, logManager);
+        InitP2P(ip);
```

Comment:
Why not inject `IShutterP2P` instead ?

Reply:
As in pass `IShutterP2P` to the `ShutterApi` constructor? I guess we could do this, and also for `IShutterEon` and `ShutterTime`. Instead I did all of the initialisation inside the API with virtual functions so the tests can override - this seems fine to me but do you think it's better to inject?

Reply:
Ok I see, so you think it's ok to leave like this for now?

Reply:
Going to resolve in separate PR